locals.tf

```
@@ -0,0 +1,131 @@
+
+  ingest_internet_proxy = data.terraform_remote_state.ingestion.outputs.internet_proxy
+  ingest_no_proxy_list  = data.terraform_remote_state.ingestion.outputs.vpc.vpc.no_proxy_list
+  internet_proxy        = local.ingest_internet_proxy.host
+  non_proxied_endpoints = join(",", local.ingest_no_proxy_list)
+
+  ucfs_london_current_domain = local.ucfs_london_domains[local.environment]
+
+  ucfs_london_ha_broker_list = [
+    "${local.ucfs_ha_broker_prefix}00.${local.ucfs_london_current_domain}",
+    "${local.ucfs_ha_broker_prefix}01.${local.ucfs_london_current_domain}",
+    "${local.ucfs_ha_broker_prefix}02.${local.ucfs_london_current_domain}"
+  ]
+
+  ucfs_london_bootstrap_servers = {
+    development = ["n/a"]                          // stubbed only
+    qa          = ["n/a"]                          // stubbed only
+    integration = local.ucfs_london_ha_broker_list //this exists on UC's end, but we do not use it as the env is stubbed as at Oct 2020
+    preprod     = local.ucfs_london_ha_broker_list
+    production  = local.ucfs_london_ha_broker_list
+  }
+
+  kafka_london_bootstrap_servers = {
+    development = local.stub_bootstrap_servers[local.environment] // stubbed
+    qa          = local.stub_bootstrap_servers[local.environment] // stubbed
+    integration = local.k2hb_data_source_is_ucfs[local.environment] ? local.ucfs_london_bootstrap_servers[local.environment] : local.stub_bootstrap_servers[local.environment]
+    preprod     = local.ucfs_london_bootstrap_servers[local.environment] // now on UCFS Staging HA
+    production  = local.ucfs_london_bootstrap_servers[local.environment] // now on UCFS Production HA
+  }
+
+  kafka_broker_port = {
+    development = local.stub_kafka_broker_port_https
+    qa          = local.stub_kafka_broker_port_https
+    integration = local.k2hb_data_source_is_ucfs[local.environment] ? local.uc_kafka_broker_port_https : local.stub_kafka_broker_port_https
+    preprod     = local.uc_kafka_broker_port_https
+    production  = local.uc_kafka_broker_port_https
+  }
+
+  kafka_consumer_truststore_aliases = {
+    development = "ucfs_ca"
+    qa          = "ucfs_ca"
+    integration = "ucfs_ca"
+    preprod     = "ucfs_ca"
+    production  = "ucfs_ca"
+  }
+
+  kafka_consumer_truststore_certs = {
+    development = "s3://${local.certificate_auth_public_cert_bucket.id}/ca_certificates/ucfs/root_ca.pem"
+    qa          = "s3://${local.certificate_auth_public_cert_bucket.id}/ca_certificates/ucfs/root_ca.pem"
+    integration = "s3://${local.certificate_auth_public_cert_bucket.id}/ca_certificates/ucfs/root_ca.pem"
+    preprod     = "s3://${local.certificate_auth_public_cert_bucket.id}/ca_certificates/ucfs/root_ca.pem"
+    production  = "s3://${local.certificate_auth_public_cert_bucket.id}/ca_certificates/ucfs/root_ca.pem,s3://${local.certificate_auth_public_cert_bucket.id}/ca_certificates/ucfs/root_ca_old.pem"
```

I think this will hit a problem when it gets to prod because the list length doesn't match kafka_consumer_truststore_aliases

```suggestion
    production  = "s3://${local.certificate_auth_public_cert_bucket.id}/ca_certificates/ucfs/root_ca.pem"
```